### PR TITLE
Add rbd deployment rbac yaml files and docs.

### LINKED ISSUE #457: Add RBACK for rbd provisioneer
Please add rbac example for  rbd provisioneer. I'm getting:
```
E1114 11:46:10.392866       1 reflector.go:201] github.com/kubernetes-incubator/external-storage/lib/controller/controller.go:412: Failed to list *v1.PersistentVolume: persistentvolumes is forbidden: User "system:serviceaccount:default:default" cannot list persistentvolumes at the cluster scope
```
on k8s 1.8 and  adding admin caps for system:serviceaccount:default:default doesn't looking reasonable.

@cofyc, @rootfs ?

### PR DESCRIPTION
**Pull request by cofyc**
**What this PR does / why we need it**:

- Add  `rbac` and `non-rbac` deployment files and deployment [docs](https://github.com/cofyc/external-storage/tree/rbd_deploy/ceph/rbd/deploy/README.md)
- Move class/secrets/claim examples files into `examples` directory

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

fixes #457.

cc @wongma7 @rootfs

### CODE REVIEW

**Comment by pastukhov** [comment]
Looks good and RBAC is working for me.

File: ceph/rbd/deploy/rbac/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    resources: ["persistentvolumeclaims"]
+    verbs: ["get", "list", "watch", "update"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["services"]
```
Comment (rootfs) [inline ceph/rbd/deploy/rbac/clusterrole.yaml]: why do we need services?

Comment (cofyc) [reply ceph/rbd/deploy/rbac/clusterrole.yaml]: @rootfs My mistake, I just copied clusterrole from gluster/block. `services` permissions are removed.

**Pull request merged**